src/components/typography/H4.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { IOColorsStatusForeground, IOTheme } from "../../core/IOColors";
+import { FontFamily, IOFontWeight } from "../../utils/fonts";
+import { useTypographyFactory } from "./Factory";
+import { ExternalTypographyProps, TypographyProps } from "./common";
+
+type AllowedColors = IOColorsStatusForeground | IOTheme["textHeading-default"];
+type AllowedWeight = Extract<IOFontWeight, "SemiBold" | "Regular">;
+
+type OwnProps = ExternalTypographyProps<
+    TypographyProps<AllowedWeight, AllowedColors>
+> & { isDesignSystemEnabled?: boolean };
+
+/* Common typographic styles */
+export const h2FontSize = 20;
+export const h2LineHeight = 24;
+
+/**
+ * Typography component to render `H4` text with font size {@link fontSize} and fontFamily {@link fontName}.
+ * default values(if not defined) are weight: `Regular/SemiBold`, color: `black/bluegreyDark` if design system is enabled or not
+ * @param props
+ * @constructor
+ */
+export const H4: React.FC<OwnProps> = props => {
+    const isDesignSystemEnabled = props.isDesignSystemEnabled ?? false;
```

Comment:
this refers to a legacy check, we need to remove and leave only the new implementation for H6